pkg/iac/scanners/terraform/parser/parser_test.go

```diff
@@ -905,6 +906,114 @@ data "http" "example" {
 
 func TestForEach(t *testing.T) {
```

Comment:
Sorry I forgot to add one thing: can we have these tests in a table instead of a bunch of `t.Run()`s? Looks like they have quite a bit of similarity between the cases that we can refactor

Reply:
Done https://github.com/aquasecurity/trivy/pull/6151/commits/6c11ee0abfbfe6e95147abb192888b68711a5c5a